nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java

```diff
@@ -122,12 +127,122 @@ public TempBlob removeMirrorUrlFromArtifactsXml(final TempBlob artifact,
     }
   }
 
-  private InputStream xmlInputStream(final TempBlob artifact, final String extension, final Path artifactsTempFile)
+  public TempBlob editUrlPathForCompositeRepository(
+      final TempBlob artifact,
+      final URI remoteUrl,
+      final Repository repository,
+      final String file,
+      final String extension)
+  {
+    try {
+      Path tempFile = createTempFile("p2-artifacts-composite-" + randomUUID().toString(), "xml");
+      // This is required in the case that the input stream is a jar to allow us to extract a single file
+      Path artifactsTempFile = createTempFile("jar-entry-" + randomUUID().toString(), "xml");
+      try {
+        try (InputStream xmlIn = xmlInputStream(artifact, file + "." + "xml", extension, artifactsTempFile)) {
+          try (OutputStream xmlOut = xmlOutputStream(file + "." + "xml", extension, tempFile)) {
```

Comment:
4 'try' blocks look very odd. Lines 142 and 143 can be combined.

Reply:
Done